webcompat/static/js/lib/bugform.js

```diff
@@ -176,25 +176,23 @@ function BugForm() {
       $("[value=" + problemType[1] + "]").click();
     }
 
-    // If we have one or more details params,
+    // If we have a details params, JSON decode it and
     // add that to the end of the steps to reproduce textarea
-    var details = location.href.match(/details=([^&]*)/g);
-    if (details !== null) {
+    var details = location.href.match(/details=([^&]*)/);
```

Comment:
I guess details is what is sent from the browser side to the form. It's a rather vague name compared to something like `browser_config` but let's fly with it given that it is not really part of this PR. I just realized now how it was called. 

```js
var url = 'https://webcompat.com/issues/new?url=https%3A%2F%2Fadobe.github.io%2FSpry%2Farticles%2Fjson_primer%2Fjson_primer.html&src=desktop-reporter&details=%5B%22gfx.webrender.all%3A+false%22%2C%22gfx.webrender.blob-images%3A+2%22%2C%22gfx.webrender.enabled%3A+false%22%2C%22image.mem.shared%3A+2%22%2C%22layout.css.servo.enabled%3A+true%22%5D&label=type-stylo'
var details = url.match(/details=([^&]*)/);
// Array [ "details=%5B%22gfx.webrender.all%3A+false%22%2C%22gfx.webrender.blob-images%3A+2%22%2C%22gfx.webrender.enabled%3A+false%22%2C%22image.mem.shared%3A+2%22%2C%22layout.css.servo.enabled%3A+true%22%5D", "%5B%22gfx.webrender.all%3A+false%22%2C%22gfx.webrender.blob-images%3A+2%22%2C%22gfx.webrender.enabled%3A+false%22%2C%22image.mem.shared%3A+2%22%2C%22layout.css.servo.enabled%3A+true%22%5D" ]
details[1]
// "%5B%22gfx.webrender.all%3A+false%22%2C%22gfx.webrender.blob-images%3A+2%22%2C%22gfx.webrender.enabled%3A+false%22%2C%22image.mem.shared%3A+2%22%2C%22layout.css.servo.enabled%3A+true%22%5D"
details[1].replace(/\+/g, "%20")
// "%5B%22gfx.webrender.all%3A%20false%22%2C%22gfx.webrender.blob-images%3A%202%22%2C%22gfx.webrender.enabled%3A%20false%22%2C%22image.mem.shared%3A%202%22%2C%22layout.css.servo.enabled%3A%20true%22%5D"
decodeURIComponent(details[1].replace(/\+/g, "%20"))
// "[\"gfx.webrender.all: false\",\"gfx.webrender.blob-images: 2\",\"gfx.webrender.enabled: false\",\"image.mem.shared: 2\",\"layout.css.servo.enabled: true\"]"
JSON.parse(decodeURIComponent(details[1].replace(/\+/g, "%20")))
// Array [ "gfx.webrender.all: false", "gfx.webrender.blob-images: 2", "gfx.webrender.enabled: false", "image.mem.shared: 2", "layout.css.servo.enabled: true" ]
JSON.parse(decodeURIComponent(details[1].replace(/\+/g, "%20"))).join("\n")
/* "gfx.webrender.all: false
gfx.webrender.blob-images: 2
gfx.webrender.enabled: false
image.mem.shared: 2
layout.css.servo.enabled: true" */
```

And all of this is added to the Steps to reproduce field. 

This works as expected.
I have seen a couple of times, people confused about why there was these things in the textarea. It probably looks like something broken.

What about feeding these to a hidden input parameter in the form and to add them to the issue body after POSTING. So people from triage see the parameters, but not reporters at the moment of filing the bug.

We keep the information, but it doesn't get into the way of reporting.

Reply:
> What about feeding these to a hidden input parameter in the form and to add them to the issue body after POSTING. So people from triage see the parameters, but not reporters at the moment of filing the bug.

Yep, I like this idea a lot. I've been wanting to do something similar for `about:support` data, but rather than appearing in the issue body, put that in a `.txt` file somewhere on the server. I'll file a follow up bug.

Reply:
> It's a rather vague name compared to something like browser_config but let's fly with it given that it is not really part of this PR.

Good point. Details originally was added to support `type-media` error messages and stack traces. So it really was more about contextual details.